homeassistant/components/fan/dyson.py

```diff
@@ -233,3 +236,11 @@ def speed_list(self: ToggleEntity) -> list:
     def supported_features(self: ToggleEntity) -> int:
         """Flag supported features."""
         return SUPPORT_OSCILLATE | SUPPORT_SET_SPEED
+
+    @property
+    def state_attributes(self: ToggleEntity) -> dict:
+        """Return optional state attributes."""
+        attributes = super().state_attributes
```

Comment:
No need call super,
```python
attributes = {}
```

or, you use just one statement
```python
return {
    ATTR_IS_NIGHT_MODE: self.is_night_mode,
    ATTR_IS_AUTO_MODE: self.is_auto_mode,
}
```

Reply:
Alright, I was confused by the typing hints, `state_attributes(self: ToggleEntity`. so, self is FanEntity, not ToggleEntity, this typing was wrong.

Reply:
Do you mean I should change it to `def state_attributes(self: FanEntity) -> dict:`? I was just following the type hint of `self` from same class and the `state_attributes`'s super class method signature

Reply:
This is so wrong, `self` should not have type hint at all.

Reply:
Does that mean the type hints in [__ init __.py](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/fan/__init__.py#L310) are wrong? I myself am not sure because these are new to me. Let me know and if so I can simply delete the type hint no problem! Thanks

Reply:
see #15732